### ISSUE TITLE
Nodes can overlap nodes sometimes

### ISSUE THREAD

**Issue by AxelRICHARD** [eclipse-sirius/sirius-web#1540]
* [x] **I have checked that this bug has not yet been reported by someone else**
* [x] **I have checked that this bug appears on Chrome**
* [x] **I have specified the version** :
- 2022.11.4
- other: please specify
* [x] **I have specified my environment** : 
- my computer: macOS
- Cloud

### Screenshots

![Capture d’écran 2022-12-22 à 14 20 42](https://user-images.githubusercontent.com/717752/209143550-d6af8092-a111-42dd-a48e-e7b6a6bcb861.png)

### Steps to reproduce

1. Create a Topography Diagram on the Robot Flow sample model.
2. Create a new Composite processor from the palette
3. Try to move the new CompositeProcessor at the same position than the screenshot above.

### Expected behavior

The overlapped Node (Capture_SubSystem) should move a little bit to avoid overlapping.

### Actual behavior

The overlapped Node (Capture_SubSystem) does not move.

**Comment by pcdavid** [comment]
This seems related to the presence of sub-nodes:

https://user-images.githubusercontent.com/10608/212730258-38834e1c-d2c2-4491-9d68-a5a0c3cbcdc4.mp4

**Comment by pcdavid** [comment]
Nope, that's not the criterion: it seems to (also?) depend on the size? aspect-ratio? of the elements...

https://user-images.githubusercontent.com/10608/212734794-239bda2a-91c4-4a10-a2ee-a8633f6bcbde.mp4
